merino-wikipedia/src/domain.rs

```diff
@@ -0,0 +1,137 @@
+//! Data types used to define the data this crate works with;
+
+use anyhow::anyhow;
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, TryFromInto};
+
+/// A page on Wikipedia. This type is a partial
+#[serde_as]
+#[derive(Deserialize, Serialize, Clone, Debug)]
+pub struct WikipediaDocument {
+    /// The human readable title of the page.
+    pub title: String,
+
+    /// The contents of the page, in Wikitext.
+    pub page_text: String,
+
+    /// The Wikipedia namespace of the page.
+    #[serde_as(as = "TryFromInto<i32>")]
+    pub namespace: WikipediaNamespace,
+
+    /// The Wikipedia ID for this page. It is expected that if two pages have
+    /// the same page_id, they are the same page, with possible edits.
+    pub page_id: u32,
+}
+
+///
```

Comment:
Do you mind adding some documents here? just in case that people are not familiar with the namespaces in ES.

Reply:
Oops, thanks for catching this.